src/main/java/jarvis/logic/parser/AddTaskCommandParser.java

```diff
@@ -30,7 +30,8 @@ public AddTaskCommand parse(String args) throws ParseException {
         }
 
         TaskDesc taskDesc = ParserUtil.parseTaskDesc(argMultimap.getValue(PREFIX_TASK_DESC).get());
-        TaskDeadline taskDeadline = ParserUtil.parseDeadline(argMultimap.getValue(PREFIX_DEADLINE).orElse(""));
+        String deadline = argMultimap.getValue(PREFIX_DEADLINE).orElse("");
+        TaskDeadline taskDeadline = deadline.equals("") ? null : ParserUtil.parseDeadline(deadline);
```

Comment:
Similar to the comment for `AddNotesCommandParser`, I think we should replace this with `argMultimap.getValue(prefix).isPresent()`